projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php

```diff
@@ -212,6 +182,44 @@ function wpcom_add_wpcom_menu_item() {
 }
 add_action( 'admin_menu', 'wpcom_add_wpcom_menu_item' );
 
+/**
+ * Add All Sites menu to the right side of the WP logo on the masterbar.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar - The WP_Admin_Bar instance.
+ */
+function add_all_sites_menu_to_masterbar( $wp_admin_bar ) {
+	if ( ! function_exists( 'wpcom_is_nav_redesign_enabled' ) || ! wpcom_is_nav_redesign_enabled() ) {
+		return;
+	}
+
+	/**
+	 * Don't show `All Sites` to administrators without a WordPress.com account being attached,
+	 * as they don't have access to any of the pages.
+	 */
+	if ( ! current_user_has_wpcom_account() ) {
+		return;
+	}
+
+	wp_enqueue_style(
+		'wpcom-site-menu-masterbar',
+		plugins_url( 'build/wpcom-site-menu/wpcom-site-menu.css', Jetpack_Mu_Wpcom::BASE_FILE ),
+		array(),
+		Jetpack_Mu_Wpcom::PACKAGE_VERSION
+	);
+
+	$wp_admin_bar->add_node(
+		array(
+			'id'    => 'all-sites',
+			'title' => __( 'All Sites', 'jetpack-mu-wpcom' ),
+			'href'  => 'https://wordpress.com/sites',
+			'meta'  => array(
+				'class' => 'wp-admin-bar-all-sites',
+			),
+		)
+	);
+}
+add_action( 'admin_bar_menu', 'add_all_sites_menu_to_masterbar', 15 );
```

Comment:
this priority comes from https://github.com/WordPress/wordpress-develop/blob/8d0aed455b1790c5d51386f7675d9e8d68e48edf/src/wp-includes/class-wp-admin-bar.php#L657-L659